clients/drcachesim/reader/file_reader.h

```diff
@@ -156,14 +156,13 @@ template <typename T> class file_reader_t : public reader_t {
         }
         VPRINT(this, 2, "Read header: ver=%zu, pid=%zu, tid=%zu\n", header.addr, pid.addr,
                tid.addr);
+        queue_to_return_next(marker_queue);
         // The reader expects us to own the header and pass the tid as
         // the first entry.
-        queue_.push(tid);
-        queue_.push(pid);
-        while (!marker_queue.empty()) {
-            queue_.push(marker_queue.front());
-            marker_queue.pop();
-        }
+        std::queue<trace_entry_t> tid_pid;
+        tid_pid.push(tid);
+        tid_pid.push(pid);
+        queue_to_return_next(tid_pid);
```

Comment:
Hmm, is it simpler to have a single queue?  s/marker_queue/readahead_queue/ and push_front pid and then tid (yes reverse but w/ only 2 it seems clear)?  Then there aren't 2 calls to return_next.